pkg/executor/sortexec/parallel_sort_worker.go

```diff
@@ -101,59 +109,69 @@ func (p *parallelSortWorker) multiWayMergeLocalSortedRows() []chunk.Row {
 		}
 		resultSortedRows = append(resultSortedRows, row)
 	}
+	p.localSortedRows = nil
 	return resultSortedRows
 }
 
-func (p *parallelSortWorker) sortBatchRows(batchRows []chunk.Row) {
-	slices.SortFunc(batchRows, p.keyColumnsLess)
-	p.localSortedRows = append(p.localSortedRows, chunk.NewIterator4Slice(batchRows))
+func (p *parallelSortWorker) sortBatchRows() {
+	slices.SortFunc(p.batchRows, p.keyColumnsLess)
+	p.localSortedRows = append(p.localSortedRows, chunk.NewIterator4Slice(p.batchRows))
+	p.batchRows = make([]chunk.Row, 0, p.maxSortedRowsLimit)
 }
 
 func (p *parallelSortWorker) sortLocalRows() []chunk.Row {
 	// Handle Remaining batchRows whose row number is not over the `maxSortedRowsLimit`
 	if len(p.batchRows) > 0 {
-		p.sortBatchRows(p.batchRows)
+		p.sortBatchRows()
 	}
 
 	return p.multiWayMergeLocalSortedRows()
 }
 
+func (p *parallelSortWorker) addChunkToBatchRows(chk *chunk.Chunk) {
+	chkIter := chunk.NewIterator4Chunk(chk)
+	row := chkIter.Begin()
+	for !row.IsEmpty() {
+		p.batchRows = append(p.batchRows, row)
+		row = chkIter.Next()
+	}
+}
+
 // Fetching a bunch of chunks from chunkChannel and sort them.
 // After receiving all chunks, we will get several sorted rows slices and we use k-way merge to sort them.
 func (p *parallelSortWorker) fetchChunksAndSort() {
+	for p.fetchChunksAndSortImpl() {
```

Comment:
Why need this function? 
Use
```
for {
    select {
....
    }
}
```
in fetchChunksAndSortImpl is ok?

Reply:
> Why need this function? Use
> 
> ```
> for {
>     select {
> ....
>     }
> }
> ```
> 
> in fetchChunksAndSortImpl is ok?

Because we can use `defer` keyword if we wrap these codes in a function and further changes in these codes could avoid to add `p.fetcherAndWorkerSyncer.Done()`.